src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -267,24 +253,7 @@ public TagHelperContent EndTagHelperWritingScope()
             }
 
             var tagHelperContent = new DefaultTagHelperContent();
-            var razorWriter = writer as RazorTextWriter;
-            if (razorWriter != null)
-            {
-                tagHelperContent.Append(razorWriter.Buffer);
-            }
-            else
-            {
-                var htmlContentTextWriter = writer as HtmlContentWrapperTextWriter;
-                if (htmlContentTextWriter != null)
-                {
-                    tagHelperContent.Append(htmlContentTextWriter.ContentBuilder);
-                }
-                else
-                {
-                    tagHelperContent.AppendHtml(writer.ToString());
-                }
-            }
-
+            tagHelperContent.Append(writer.ContentBuilder);
```

Comment:
Might be a negligibly small optimization but would be nice if we could pass the ContentBuilder into the ctor of the `DefaultTagHelperContent`. This would help avoid the backing `List<IHtmlContent>` from resizes.